main.go

```diff
@@ -135,6 +135,10 @@ func main() {
 	logrus.SetLevel(level)
 	log.EnableTracing(true)
 	log.FromContext(ctx).WithField("duration", time.Since(now)).Infof("completed phase 1: get config from environment")
+	logruslogger.SetupLevelChangeOnSignal(ctx, map[os.Signal]logrus.Level{
+		syscall.SIGUSR1: logrus.TraceLevel,
+		syscall.SIGUSR2: level,
+	})
```

Comment:
I tihnk we can ignore it if current lelvel is trace.

Reply:
I think adding a new 'if branch' is lesser evil than having extra goroutine and subscription on the signals.

Reply:
okay, you convinced me.

Reply:
Also, I agree that it would be better to not add a new if branch; probably we could add a check on signal map size in the 'SetupLevelChangeOnSignal' and say if size is less than 2, then do nothing. Thoughts?

Reply:
:'D how about we add it to the function in the sdk instead? :'D
`run golangci-lint
  Running [/home/runner/golangci-lint-1.53.3-linux-amd64/golangci-lint run --out-format=github-actions --timeout 3m --verbose] in [] ...
  Error: cyclomatic complexity 16 of func `main` is high (> 15) (gocyclo)`

Reply:
> Also, I agree that it would be better to not add a new if branch; probably we could add a check on signal map size in the 'SetupLevelChangeOnSignal' and say if size is less than 2, then do nothing. Thoughts?

Yeah sounds good

Reply:
I mean yeah in addition to checking if the original level is not trace, in addition to that it is a correct opinion to check the map's size too yeah.